Firestore/Example/Tests/Local/StringViewTests.mm

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <leveldb/slice.h>
+#import "Firestore/Source/Local/StringView.h"
+
+#import <XCTest/XCTest.h>
+
+using Firestore::StringView;
+
+@interface StringViewTests : XCTestCase
+@end
+
+@implementation StringViewTests
+
+- (void)testStringViewNSStringToSliceWithUSAscii {
+  NSString *usAsciiChars = @"abcdefg ABCDEFG 12345 !@#$%";
+  [self convertNSStringToStringViewAndBackAndEnsureEqual:usAsciiChars];
+}
+
+- (void)testStringViewNSStringToSliceWithNonUSAscii {
+  NSString *nonUsAsciiChars = @"ó¹";
+  [self convertNSStringToStringViewAndBackAndEnsureEqual:nonUsAsciiChars];
+}
+
+- (void)convertNSStringToStringViewAndBackAndEnsureEqual:(NSString *)str {
```

Comment:
You might consider making this a macro because as it stands this will report assertion failures with the file and line of the assertion in here rather than in the tests themselves.

cf https://github.com/firebase/firebase-ios-sdk/blob/4fbc96904fe86b3a0cdb91bfa6d96491221eb6b9/Firestore/Example/Tests/Util/FSTComparisonTests.m#L26

Reply:
I'm not sure I understand the concern? If I undo the fix and rerun the tests, xcode correctly reports the second test as broken. (The line number is of course within the helper function, but that seems reasonable to me?)

Reply:
The concern is minimal but it's a point of consistency and understanding the tools. The general advice is to make custom assertions macros in XCTest because it leads to a better result.

Consider [this test](https://github.com/firebase/firebase-ios-sdk/blob/4fbc96904fe86b3a0cdb91bfa6d96491221eb6b9/Firestore/Example/Tests/Util/FSTComparisonTests.m#L67): there's a series of assertions in there. By making these assertions macros it's clear not just that the test failed at all, but which assertion in the test failed.

Unlike a crash where we get a stack trace, for test failures we only get a single file and line number.

Also, unlike GoogleTest or JUnit, XCTest continues after assertion failures by default so in a test like the one above after a failure you'd end up with 10 red dots all on the same line.

Reply:
done.